lib/checks/keyboard/banner-is-top-level.js

```diff
@@ -0,0 +1,19 @@
+const landmarks = axe.commons.aria.getRolesByType('landmark');
```

Comment:
This is basically a repeat of `main-is-top-level.js`. I suggest having these 3 checks point to the same evaluate file. You can change what element they test through the rule selector as well as the check.json options property. So you would have `main-is-top-level.json`, `banner-is-top-level.json` and `contentinfo-is-top-level.json` each point to `landmark-is-top-level.js`, and pass in a different `options` object which tells it what element you're trying to test.

Reply:
Not all landmarks need to be top level, only main, banner and contentinfo should be top level.  In many cases like region landmarks, they will identify subsections of "named" landmarks.  Some assistive technologies support navigation of the parent child relationships of landmarks.

See ARIA Authoring Practices 1.1 for landmarks:
https://www.w3.org/TR/wai-aria-practices-1.1/#aria_landmark

Reply:
But what you seem to be saying is just have one rule, that says banner, contentinfo and main landmarks must be top level, so we will look at combining the three rules into one rule.

Reply:
My main concern is that the messaging might not be as clear.  Since people may think they can only have one MAIN, CONTENTINFO or BANNER, when they can have one of each or they can all be top level

Reply:
My main concern is messaging in a more general rule will not be as clear, since the message will talk about all three landmarks, and some pages may one have a MAIN landmark

Reply:
I want to make sure you've seen #681 as part of this conversation

Reply:
@jongund No no, please don't merge the rules. I think three rules makes the most sense. What I'm saying is that these three checks should have a shared `evaluate` function. Have a look here:
https://github.com/dequelabs/axe-core/tree/develop/lib/checks/mobile

Here are two checks, `meta-viewport` and `meta-viewport-large` that are very similar. Instead of each of these having it's own `.js` file, they both use `meta-viewport-scale.js`. This way we don't have two versions of this `.js` file that do essentially the same thing.

Reply:
@WilcoFiers OK  We can do that, I don't think we need the options property set, since we can use the tag information from the rules json file to determine which landmark is being tested

Reply:
As far as I know the check doesn't have access to what rule is running. But if you can find a good way to do it without options that's fine by me. :)